python/amici/ode_export.py

```diff
@@ -833,17 +834,24 @@ def import_from_sbml_importer(self,
             imported SBML model
         """
 
+        # get symbolic expression from SBML importers
         symbols = copy.copy(si.symbols)
 
-        # setting these equations prevents native equation generation
-        self._eqs['dxdotdw'] = si.stoichiometric_matrix
-        self._eqs['w'] = si.flux_vector
-        self._syms['w'] = sp.Matrix(
-            [sp.Symbol(f'flux_r{idx}', real=True)
-             for idx in range(len(si.flux_vector))]
-        )
-        self._eqs['dxdotdx'] = sp.zeros(si.stoichiometric_matrix.shape[0])
-
+        # assemble fluxes and add them as expressions to the model
```

Comment:
I added a comment here, that this generates a preliminary dxdt.
Seems like that was erased. Please leave it in. It not really understandable otherwise why we generate the dynamics twice (and overwrite what we just created). That caused me some headache to undertsand and it will cause headaches to the next person working on it, if not documented

Reply:
But we do not generate a preliminary `dxdt` here? I am not sure what you think is generated twice.

Reply:
We use the fluxes to create ` symbols['species']['dt']`.
1. If you're not familiar with the routine, you would expect that this should basically be teh symbolic expression for dxdot.
2. Based on ` symbols['species']['dt']`, model quantities are created. Based on those quantities, it's decied, whether a state is marked as constant or not, which later on influences CL computation. In this sense, we indeed create information on xdot already here. If you're not really, really familiar with ode_export.py, there is no way that would understand how these code parts are intertwined...

Reply:
Maybe to make it more precise: Based on what is created here, ode_model._states[*index of state*]['dt'] is written.
However, ode_model._sym['x'] is not created from it. Before, those two qunatities (ode_model._states[*index of state*]['dt'] and  ode_model._sym['x']) were in line, now they aren't any more (and can't be due to CLs).
That's really not intuitive and took me quite a while. We need to document that somehow, e.g., by a comment here.

Reply:
the values in `symbols['species']['dt']` use the public API of `State`. There is nothing that is created twice. Maybe its confusing to have `import_from_sbml_importer` as member of `SbmlImporter` as thats how it is intended to operate wrt `ODE_model`.

Don't forget that `ode_model._states` is a dict, i.e. dont assume anything about order. `import_from_sbml_importer` is super hacky and makes a shitton of assumptions, such as the ordering and number of elements in `ode_model._sym['x']` and `ode_model._states[index of state]`. If you want the ordering of two things to be the same order, put them in a list of dicts/tuples/whatever. If they are in two seperate lists, assume thats intential that they may not have the the same order. just because they the same order now, doesn't mean they will have it in the future. This may not always be the case everywhere as some amici code is pretty old by now, but thats how things are supposed to be (not just in amici, but generally in python code)

In this part of the code, you shouldn't even think or crae about `ode_model._states`. It doesn't appear anywhere in this function, and there shouldn't be any mention/documentation about it. 

In terms of the circular dependency that you are referring to, the important thing is that `process_conservation_laws(self)` requires that species are already added to the model, so it needs to be done after those are added to the model.